### PR TITLE
Form multiselect

### PR DESCRIPTION
**Pull request by antoniotrkdz**
Description as per commit.
It incorporates the changes from the previous one so once it get merged to master I hope the size of this PR will be OK.

### CODE REVIEW

**Comment by des-des** [comment]
@majakudlicka I am still waiting for the previous pull to go in

**Comment by majakudlicka** [comment]
Gotcha, sorry!

File: react-ui/src/components/Form_Post_Job.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import * as actions from '../actions/post_job.js';
 import {connect} from 'react-redux';
 import {Router, Route, IndexRoute, browserHistory} from 'react-router';
 import isEmpty from 'lodash/isEmpty';
+import Categories from '../constants/job_categories.js';
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: Lower case please, if you want you can make it all caps, or make the keys all caps

File: react-ui/src/components/Form_Post_Job.js [hunk 2]
```diff
@@ -143,9 +144,19 @@ class Form_Post_Job extends Component {
...
+                    {elem}
+                </option>
+            );
+        });
+        
         if (!this.state) {
             return <div>Loading</div>;
         }
+        
+        console.log(this.state);
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: :scream:

File: react-ui/src/components/Form_Register_Student.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
 import React, {Component} from 'react';
 import LoadingButton from './LoadingButton.js';
 import ErrorMessage from './ErrorMessage.js';
+import Categories from '../constants/job_categories.js';
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: same here, lower case i think

File: react-ui/src/components/Form_Register_Student.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 
 class Form_Register_Student extends Component {
     constructor() {
         super();
         this.onSubmit = this.onSubmit.bind(this);
         this.onChange = this.onChange.bind(this);
+        this.onChangeJobCat = this.onChangeJobCat.bind(this);
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: Where is the cat?

I do not think there are any cats

File: react-ui/src/components/Form_Register_Student.js [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
         this.checkEmail = this.checkEmail.bind(this);
 
         this.state = {
-            student: {},
+            student: {
+                jobCat: {},
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: category, not cat. (that is an animal)

File: react-ui/src/components/Form_Register_Student.js [hunk 6]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+                    {elem}
+                </option>
+            );
+        });
+
+        if (!this.state) {
+            return <div>Loading</div>;
+        }
+        
+        console.log(options);
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: no

File: react-ui/src/components/Form_Register_Student.js [hunk 7]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+        console.log(this.state);
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: no

File: react-ui/src/components/Form_Register_Student.js [hunk 8]
```diff
@@ -220,97 +249,87 @@ class Form_Register_Student extends Component {
...
                         className="form-control"
-                        name="index1"
+                        name="cat1"
                         id="jobCategory1"
-                        type="text"
+                        type="dropdown"
                         placeholder="Select"
-                        value={this.state.student.index1}
-                        onChange={this.onChange}
+                        value={this.state.student.jobCat.cat1}
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: this is the first cat?

File: react-ui/src/components/Form_Register_Student.js [hunk 9]
```diff
@@ -220,97 +249,87 @@ class Form_Register_Student extends Component {
                     </label>
                     <input
                         className="form-control"
-                        name="index1"
+                        name="cat1"
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: why is this input a cat?

File: react-ui/src/components/pages/SearchStudents.js [hunk 10]
```diff
@@ -2,6 +2,7 @@ import React, {Component} from 'react';
 import {connect} from 'react-redux';
 import * as actions from '../../actions/search_students.js';
 import {Link} from 'react-router';
+import Categories from '../../constants/job_categories.js';
```
Comment (des-des) [inline react-ui/src/components/pages/SearchStudents.js]: naming

File: react-ui/src/components/pages/SearchStudents.js [hunk 11]
```diff
@@ -67,18 +76,8 @@ class BrowseStudents extends Component {
...
-                            <option value="Tutoring- Mathematics" />
-                            <option value="Cat Sitting" />
-                            <option value="Plant watering" />
-                            <option value="Babysitting" />
-                            <option value="Cooking" />
-                            <option value="House Cleaning" />
-                            <option value="Band playing" />
-                            <option value="photography" />
-                            <option value="Other" />
+                            <option value="" disabled />
```
Comment (des-des) [inline react-ui/src/components/pages/SearchStudents.js]: hmm there is not click / onChnage handler / how is state managed here, this is confusing

File: react-ui/src/components/Form_Post_Job.js (hunk 1 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: Done.

File: react-ui/src/components/Form_Post_Job.js (hunk 2 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 3 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 6 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done

File: react-ui/src/components/Form_Register_Student.js (hunk 7 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/pages/SearchStudents.js (hunk 10 above)
Comment (antoniotrkdz) [inline react-ui/src/components/pages/SearchStudents.js]: done.

File: react-ui/src/components/Form_Register_Student.js (hunk 4 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 5 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: done.

File: react-ui/src/components/Form_Register_Student.js (hunk 9 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: Done.

File: react-ui/src/components/Form_Register_Student.js (hunk 8 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: done

File: react-ui/src/components/pages/SearchStudents.js (hunk 11 above)
Comment (antoniotrkdz) [inline react-ui/src/components/pages/SearchStudents.js]: That has been changed? no?

Comment (antoniotrkdz) [inline react-ui/src/components/pages/SearchStudents.js]: Ahh, sorry. I think it is managed by the imput field above.
                        ```<input
                            className="form-control"
                            id="Browse Students"
                            type="text"
                            placeholder="Browse Students"
                            list="students"
                            onChange={this.onStudentSearchChange}
                            value={this.props.searchTerm}
                        />```

**Comment by antoniotrkdz** [comment]
Sorry I made the same change to a file that I forgot before, but it is the same modification.

**Pull request merged**